### DISCUSSION TITLE
pgcli on linux server without internet connection

### DISCUSSION THREAD

**Discussion by bat-manav** [dbcli/pgcli#1348] [Q&A] [answered]
Hi,  I am trying to install pgcli on my RHEL server ( on which i dont have internet connection).

I downloaded the tar file and unzipped on my server and ran python setup.py install. But it is failing , as it is trying to download some dependencies.

Is there any list of dependencies that we can install for pgcli utility.

![Screenshot 2022-06-13 at 9 24 03 PM](https://user-images.githubusercontent.com/68768446/173394468-8871e307-abaf-4ceb-abe2-0ade58182238.png)

**Comment by j-bennet** [comment, marked as answer]
@bat-manav,

It looks like you have more problems than just the dependencies. You're getting syntax errors because you're using Python 2.7, which reached end-of-life on Jan 1, 2020. As of version 3.0.0, we dropped support of Python 3.5 and older.

I don't know what version of RHEL you're using, but AFAIK, RHEL 8 ships with python 3.6. You probably have `python3` and `pip3` commands on your system. To check, try `python3 --version` and `pip3 --version`.

If you have `python3`, and it's 3.6+, you should be able to install `pgcli` with `python3 setup.py`.

If all you have is Python 2.7, and you're unable to install Python 3.6+, you could download and install `pgcli==2.2.0`. It's the last version that still has support for Python 2.7.

As for the dependencies, here is an article that describes how to download dependencies of a Python package for offline use:

https://www.activestate.com/resources/quick-reads/how-to-download-python-dependencies/
